### PR TITLE
#983 Update vaccine expansion logic

### LINKED ISSUE #983: Vaccine module extra logic in requisition is incomplete
Build Number: v2.2.0 rc 0

Description:

Currently there is an expansion in requisitions for vaccine items. This expansion shows details such as open vial wastage, doses given etc.

This should use the data from the last requisition (?) but currently, uses data from `1/1/2000`. Also, the `catchment polutation` field is a constant `100` and should be removed.

Reproducible:

N/A

Reproduction Steps:

Create a requisition with a vaccine item, view the expansion

Comments:

Fields that should be shown

- Last requisition date
- Doses given
- Open vial wastage
- Close vial wastage

### PR DESCRIPTION
**Pull request by wlthomson**
Fixes #983.

Open and closed vial wastage is calculated from CIs and SCs (inventory adjustments) since latest requisition for given vaccine item. If no such requisition exists, no date constraints are applied.

Test file: [vaccines.zip](https://github.com/openmsupply/mobile/files/3210088/vaccines.zip).

### CODE REVIEW

File: src/pages/SupplierRequisitionPage.js [hunk 1]
```diff
@@ -379,45 +376,45 @@ export class SupplierRequisitionPage extends React.Component {
...
+    const isPrevRequisition = prevRequisitions.length > 0;
+
+    const prevRequisitionDate = isPrevRequisition ? prevRequisitions[0].entryDate : null;
+
+    // Get transactions for this item since most recent requisition.
 
-    const openVialWastage = transactionBatchesCI.sum('numberOfPacks') * dosesInVial - doses;
+    const transactionBatches = database.objects('TransactionBatch').filtered('itemId = $0', requisitionItem.item.id);
+          
+    const recentTransactionBatches = prevRequisitionDate ? transactionBatches.filtered('transaction.confirmDate >= $0', prevRequisitionDate) : transactionBatches;
```
Comment (joshxg) [inline src/pages/SupplierRequisitionPage.js]: Quite a few of these lines are to long and this has been unformatted according to our rules!

**Comment by andreievg** [comment]
@wlthomson can you please follow up, cheers

**Comment by wlthomson** [comment]
> @wlthomson can you please follow up, cheers

This is on my backlog, it will be done by tonight.

**Comment by wlthomson** [comment]
> Awesome, thanks @wlthomson.
> Can you also remove catchment population

I thought I had done this, did I miss something?

File: src/pages/SupplierRequisitionPage.js (hunk 1 above)
Comment (wlthomson) [reply src/pages/SupplierRequisitionPage.js]: Whoops, I took prettier off `ctrl-s` because it kept fighting with the lint rules. Fixed by f8f8dfd1.

**Comment by wlthomson** [comment]
> What rules were you having trouble with? Can you update #845 with them?

It was all indentation related, in particular for JSX. I've updated #845.

**Comment by wlthomson** [comment]
@andreievg @joshxg Just bumping this as approved but not merged. Not sure as to current vaccine progress etc. Let me know 😄 .

**Pull request merged**